solr/core/src/java/org/apache/solr/handler/admin/RequestApplyUpdatesOp.java

```diff
@@ -67,6 +69,11 @@ public void execute(CoreAdminHandler.CallInfo it) throws Exception {
             SolrException.ErrorCode.SERVER_ERROR, "Could not apply buffered updates", e);
     } finally {
       if (it.req != null) it.req.close();
+      // NOTE: `req` must be closed _before_ core, because `req` can hold a reference to
+      // SolrIndexSearcher which can in some cases block `core.close()` from completing
+      // successfully (by holding a reference to a core Directory). Hence we cannot trivially
+      // use try-with-resources to close `core`.
```

Comment:
I think I'd use nested `try w/ resources` for SolrCore and inner `try/finally` for `it.req` that would force the order as well and make it more clear?

Reply:
+1 Kevin; we agree!